### ISSUE TITLE
Failed to create schema when the graphql files have comments

### ISSUE THREAD

**Issue by reidem** [nicolasdao/schemaglue#8]
I split my schema in four different files, when i try run my app if there is any comment in the files i get the following error:

/usr/src/app/node_modules/graphql/language/parser.js:152
throw unexpected(lexer);
	^
 GraphQLError: Syntax Error: Unexpected Name "\_\_Hke9IXDhM__"
     at syntaxError (/usr/src/app/node_modules/graphql/error/syntaxError.js:24:10)
     at unexpected (/usr/src/app/node_modules/graphql/language/parser.js:1322:33)
     at parseDefinition (/usr/src/app/node_modules/graphql/language/parser.js:152:9)
     at parseDocument (/usr/src/app/node_modules/graphql/language/parser.js:110:22)
     at Object.parse (/usr/src/app/node_modules/graphql/language/parser.js:38:10)
     at buildSchemaFromTypeDefinitions (/usr/src/app/node_modules/graphql-tools/dist/schemaGenerator.js:126:33)
     at _generateSchema (/usr/src/app/node_modules/graphql-tools/dist/schemaGenerator.js:48:18)
     at makeExecutableSchema (/usr/src/app/node_modules/graphql-tools/dist/schemaGenerator.js:64:20)
     at Object.<anonymous> (/usr/src/app/routes/index.js:26:26)
     at Module._compile (module.js:660:30)
     at Object.Module._extensions..js (module.js:671:10)
     at Module.load (module.js:573:32)
     at tryModuleLoad (module.js:513:12)
     at Function.Module._load (module.js:505:3)
     at Module.require (module.js:604:17)
     at require (internal/module.js:11:18)
     at Object.<anonymous> (/usr/src/app/app.js:7:15)
     at Module._compile (module.js:660:30)
     at Object.Module._extensions..js (module.js:671:10)
     at Module.load (module.js:573:32)
     at tryModuleLoad (module.js:513:12)
     at Function.Module._load (module.js:505:3)

I supposed there is some problem in the replaceTokens function that is not replacing the tokens with the proper comment.
Is there any reason why you need to remove the comments and restore them later?

__Modules__:
1. "apollo-server-express": "1.3.2",
2. "graphql": "^0.13.2",
3. "graphql-tools": "2.16.0",
4.  "schemaglue": "^2.0.1"

Not sure if you need more info so feel free to ask for more.

PS: Sorry for not submitting a PR but i'm starting with javascript and i didnt feel comfortable modifying other's code yet

**Comment by reidem** [comment]
### graphql/schema/sample1_schema.graphql

```
type Query {
    #Here is the trouble
    Hello(name: String!): Hello
}

#And here too
type schema {
    Query: Query
}
```

### graphql/schema/sample2_schema.graphql
```
type Hello {
    result: String!
}
```

### grapql/resolvers.js
```
const resolvers = {
    Query: {
        Hello: (root, data, context) => {return {result: `Hello ${data.name}`}}
    }

};

module.exports = resolvers;
```

### index.js
```
const express = require("express");
const router = express.Router();

const {graphqlExpress} = require('apollo-server-express');
const {graphiqlExpress} = require('apollo-server-express');

const resolvers = require("./graphql/resolvers");
const {makeExecutableSchema} = require('graphql-tools');
const glue = require('schemaglue');
const { schema, resolver } = glue('./graphql/schema');


const schemaExecutable = makeExecutableSchema({
    typeDefs: schema,
    resolvers:resolvers
});

router.use("/graphiql", graphiqlExpress({
    endpointURL: '/api'
}));

router.use(
    "/api",
    graphqlExpress( req => { return {
        schema: schemaExecutable,
        context: {req}
        }
    })
);

module.exports = router;
```
### app.js
```
const http = require("http");
const express = require("express");
const bodyParser = require("body-parser");
const index = require("./index");


const app = express();

app.use("/", index);
app.use(bodyParser.json({limit: '10mb'}));

const server = http.createServer(app);
server.listen(8000);

module.exports = {
    server: server,
    app: app
};
```
### package.json
```
{
  "name": "sample",
  "version": "1.0.0",
  "description": "",
  "main": "app.js",
  "scripts": {
    "start": "node app.js"
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "apollo-server-express": "1.3.2",
    "body-parser": "1.18.1",
    "express": "4.15.4",
    "graphql": "0.13.2",
    "graphql-tools": "2.16.0",
    "schemaglue": "2.0.1"
  }
}
```

If you delete the comments in the `sample1_schema.graphql` it works fine

**Comment by nicolasdao** [comment]
Hi @reidem ,

The issue is not related to the comments, but the inappropriate definition of the `type schema` in your **graphql/schema/sample1_schema.graphql** file. SchemaGlue takes care of the type schema itself, which means that if you define it again, there is an issue. Here is what your  **graphql/schema/sample1_schema.graphql** file should look like:

```
type Query {
    #Here is the trouble
    Hello(name: String!): Hello
}
```

That being said, I understand why you did that mistake. I will manage that situation in upcoming versions.

Thanks a lot for your awesome feedback and your support.

Cheers,

Nic

**Comment by nicolasdao** [comment]
Actually, my bad @reidem ,

I've found another issue with comments which makes your issue completely legitimate. I'm fixing that now.

**Comment by nicolasdao** [comment]
Hi @reidem ,

I've just pushed a fix. Install the new release (i.e., 2.0.2) and that should be fixed.

Thank you heaps for finding that issue and helping me making that project better.

Cheers,

Nic

**Comment by DavidRuby** [comment]
i have an issue:

/home/pi/graphql-realestate/node_modules/graphql/language/parser.js:157
    throw this.unexpected();
    ^
GraphQLError: Syntax Error: Unexpected Name "imput"
    at syntaxError (/home/pi/graphql-realestate/node_modules/graphql/error/syntaxError.js:15:10)
    at Parser.unexpected (/home/pi/graphql-realestate/node_modules/graphql/language/parser.js:1463:41)
    at Parser.parseDefinition (/home/pi/graphql-realestate/node_modules/graphql/language/parser.js:157:16)
    at Parser.many (/home/pi/graphql-realestate/node_modules/graphql/language/parser.js:1518:26)
    at Parser.parseDocument (/home/pi/graphql-realestate/node_modules/graphql/language/parser.js:111:25)
    at parse (/home/pi/graphql-realestate/node_modules/graphql/language/parser.js:36:17)
    at buildSchema (/home/pi/graphql-realestate/node_modules/graphql/utilities/buildASTSchema.js:462:43)
    at Object.<anonymous> (/home/pi/graphql-realestate/app.js:15:13)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
[nodemon] app crashed - waiting for file changes before starting...

the app failed to compile, do i have to troubleshoot everything it says there? this is above my paygrade lol

**Comment by DavidRuby** [comment]
> Hi @reidem ,
> 
> I've just pushed a fix. Install the new release (i.e., 2.0.2) and that should be fixed.
> 
> Thank you heaps for finding that issue and helping me making that project better.
> 
> Cheers,
> 
> Nic

/home/pi/graphql-realestate/node_modules/graphql/language/parser.js:157
    throw this.unexpected();
    ^
GraphQLError: Syntax Error: Unexpected Name "imput"
    at syntaxError (/home/pi/graphql-realestate/node_modules/graphql/error/syntaxError.js:15:10)
    at Parser.unexpected (/home/pi/graphql-realestate/node_modules/graphql/language/parser.js:1463:41)
    at Parser.parseDefinition (/home/pi/graphql-realestate/node_modules/graphql/language/parser.js:157:16)
    at Parser.many (/home/pi/graphql-realestate/node_modules/graphql/language/parser.js:1518:26)
    at Parser.parseDocument (/home/pi/graphql-realestate/node_modules/graphql/language/parser.js:111:25)
    at parse (/home/pi/graphql-realestate/node_modules/graphql/language/parser.js:36:17)
    at buildSchema (/home/pi/graphql-realestate/node_modules/graphql/utilities/buildASTSchema.js:462:43)
    at Object.<anonymous> (/home/pi/graphql-realestate/app.js:15:13)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
[nodemon] app crashed - waiting for file changes before starting...
